components/locale-chooser.js

```diff
@@ -121,48 +188,57 @@ function localeToHref(locale) {
   return 'https://github.com/kentcdodds/glamorous-website/blob/master/other/CONTRIBUTING_DOCUMENTATION.md'
 }
 
-function getLocaleAndHost() {
-  const locale = process.env.LOCALE
+function getHost(currentLocale) {
   const {host} = window.location
-  // eslint-disable-next-line no-unused-vars
   const [localePart, ...rest] = host.split('.')
-  if (supportedLocales.includes(locale) || localePart !== locale) {
-    return {locale, host: rest.join('.')}
-  } else {
-    return {locale: fallbackLocale, host}
+  if (localePart === currentLocale) {
+    return rest.join('.')
   }
+
+  return host
 }
 
-function mapLocale(key = fallbackLocale) {
```

Comment:
I really dislike switch statements. They're overly verbose. Could we go back to the object map?